sourcecode/shared/src/main/scala/sourcecode/SourceContext.scala

```diff
@@ -26,7 +26,12 @@ object Name extends SourceCompanion[String, Name](new Name(_)){
     var owner = Compat.enclosingOwner(c)
     while(Util.isSynthetic(c)(owner)) owner = owner.owner
     val simpleName = Util.getName(c)(owner)
-    c.Expr[sourcecode.Name](q"""${c.prefix}($simpleName)""")
+    //if the class name is anonymous, then we dig up the instance value name instead
+    val valName = if ((simpleName == "$anon") && (owner.owner.isTerm)) {
```

Comment:
Should this logic maybe be in Util.getName?

Reply:
Is it that generic?

Reply:
My bad, I meant I think case can be added to `Util.isSynthetic`. The convention is that synthetic names contain dollar signs so you might be safe to use `name.contains("$")` instead. That should handle `$anonfun` and some other related cases.